orbital_mouse/orbital_mouse.c

```diff
@@ -307,14 +367,21 @@ void housekeeping_task_orbital_mouse(void) {
     if (state.slow) {
       speed = ((uint16_t)speed) * (1 + (uint16_t)SLOW_MOVE_FACTOR_Q_8) >> 8;
     }
+    else if (state.fast) {
+      speed = ((uint16_t)speed) * (1 + (uint16_t)FAST_MOVE_FACTOR_Q4_4) >> 4;
+    }
 
     state.x -= state.move_dir * scaled_sin(speed, state.angle >> 8);
     state.y -= state.move_dir * scaled_cos(speed, state.angle >> 8);
     active = true;
   }
-
-  // Update heading angle if steering.
-  if (state.steer_dir) {
+  // Force the angle if angle is overridden
+  if (state.override_angle) {
+    set_orbital_mouse_angle_fractional(state.angle);
```

Comment:
`override_angle` and the `if` that's been added here isn't needed. The line `set_orbital_mouse_angle_fractional(state.angle)` is a no-op, setting `state.angle` to its current value.

To snap the angle, it is enough to assign `state.angle = new_angle`, as you've done in `process_record_orbital_mouse()`. To freeze the steering keys, you can add `state.steer_dir = 0` when snapping the angle.